java/java.source.base/test/unit/src/org/netbeans/modules/java/source/save/FormatingTest.java

```diff
@@ -2205,6 +2205,93 @@ public void run(WorkingCopy workingCopy) throws java.io.IOException {
         preferences.putBoolean("indentCasesFromSwitch", true);
 
     }
+
+    public void testSwitchWithTryExpr() throws Exception {
+        testFile = new File(getWorkDir(), "Test.java");
+        TestUtilities.copyStringToFile(testFile, "");
+        FileObject testSourceFO = FileUtil.toFileObject(testFile);
+        DataObject testSourceDO = DataObject.find(testSourceFO);
+        EditorCookie ec = (EditorCookie) testSourceDO.getCookie(EditorCookie.class);
+        final Document doc = ec.openDocument();
+        doc.putProperty(Language.class, JavaTokenId.language());
+
+        String content
+                = "package hierbas.del.litoral;\n\n"
+                + "public class Test {\n\n"
+                + "    public void taragui(int i) {\n"
+                + "        switch (i) {\n"
+                + "        case 0:\n"
+                + "        try {\n"
+                + "        System.out.println(i);\n"
+                + "        } catch (Exception ex) {\n"
+                + "        System.out.println(ex);\n"
+                + "        }\n"
+                + "        break;\n"
+                + "        default:\n"
+                + "        System.out.println(\"DEFAULT\");\n"
+                + "        }\n"
+                + "    }\n"
+                + "}\n";
+
+        String golden
+                = "package hierbas.del.litoral;\n\n"
+                + "public class Test {\n\n"
+                + "    public void taragui(int i) {\n"
+                + "        switch (i) {\n"
+                + "            case 0:\n"
+                + "                try {\n"
+                + "                    System.out.println(i);\n"
+                + "                } catch (Exception ex) {\n"
+                + "                    System.out.println(ex);\n"
+                + "                }\n"
+                + "                break;\n"
```

Comment:
would another case ( `case 1:...`) with try-with resource make sense?

Reply:
an arrow based switch for regression test would be awesome too! It appears to not have this issue since it must have code blocks `{}` if there is an try.
e.g:
```java
        String s = "";
        switch (s) {
            case "a" -> {
                try {
                    System.out.println("a");
                } catch (RuntimeException ex) {
                    ex.printStackTrace();
                }
            }
            case "b" ->
                System.out.println("b");
            default ->
                throw new AssertionError();
        }
```

Reply:
I had all these in the file I was debugging with, and can look into adding them.  This file will be so much nicer when we can finally use text blocks!

Reply:
I've added the try-with-resources, and also a finally and try in default.

I've left out switch with `->` for now as I'm not entirely sure whether we need source version checks any more - they're elsewhere in the test file.

Reply:
awesome, thanks!